### PR TITLE
feat: improve the COPY filting copied files performance

### LINKED ISSUE #8574: bug: `COPY INTO` CPU load takes a long time to rise
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

v0.8.89-nightly

### What's Wrong?

```
COPY INTO import FROM 's3://xx/xx/100000-100/' credentials=(aws_key_id='xx' aws_secret_key='xx') pattern ='.*[.]gz' file_format = (type = 'tsv' compression
= GZIP) force=true;
Query OK, 0 rows affected (2 min 22.85 sec)
```
The s3 path /100000-100/* only has 100 files.

At the beginning of the command execution, I observed a very low cpu load:
image

It took about 60 seconds for the CPU load to rise:
image

i checked info logs during the period of low CPU load.
image
got alot of this
```
2022-11-01T06:47:01.463339Z  INFO opendal::services::s3::backend: backend build started: Builder
2022-11-01T06:47:01.463371Z  INFO opendal::services::s3::backend: backend use root
2022-11-01T06:47:02.067443Z  INFO opendal::services::s3::backend: backend build finished: Builder
2022-11-01T06:47:01.463339Z  INFO opendal::services::s3::backend: backend build started: Builder
2022-11-01T06:47:01.463371Z  INFO opendal::services::s3::backend: backend use root
2022-11-01T06:47:02.067443Z  INFO opendal::services::s3::backend: backend build finished: Builder
2022-11-01T06:47:01.463339Z  INFO opendal::services::s3::backend: backend build started: Builder
2022-11-01T06:47:01.463371Z  INFO opendal::services::s3::backend: backend use root
2022-11-01T06:47:02.067443Z  INFO opendal::services::s3::backend: backend build finished: Builder
...
```

### How to Reproduce?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Reuse the StageFile info when filtering the copied files.

List 300 files and do force copy:
```
--rows: 1000000, size: 100*43MB
mysql> COPY into bendlog.log from @log pattern='.*[.]csv$' FILE_FORMAT = (TYPE = 'CSV'  FIELD_DELIMITER = '\t' RECORD_DELIMITER = '\n' SKIP_HEADER=1 COMPRESSION=AUTO) force=true;
Query OK, 0 rows affected (17.80 sec)

--rows: 1000000, size: 100*23MB
mysql> COPY into bendlog.log from @log pattern='.*[.]csv.gz$' FILE_FORMAT = (TYPE = 'CSV'  FIELD_DELIMITER = '\t' RECORD_DELIMITER = '\n' SKIP_HEADER=1 COMPRESSION=AUTO) force=true;
Query OK, 0 rows affected (15.68 sec)
```

Fixes #8574

### CODE REVIEW

**Comment by Xuanwo** [comment]
We can eliminate the extra stat here by reusing the `StageFileInfo` we already got during list.

**Comment by BohuTANG** [comment]
> We can eliminate the extra stat here by reusing the `StageFileInfo` we already got during list.

That's cool, and if you have an idea, I plan to continue this parallel after your PR.
From my test, this parallel will hang, I will address them tomorrow :/

**Comment by Xuanwo** [comment]
> That's cool, and if you have an idea, I plan to continue this parallel after your PR.

I will implement this tomorrow~

**Comment by BohuTANG** [comment]
@Xuanwo Done

File: src/query/service/src/interpreters/interpreter_copy_v2.rs [hunk 1]
```diff
@@ -98,14 +99,13 @@ impl CopyInterpreterV2 {
...
-    async fn filter_duplicate_files(
+    async fn filter_have_copied_files(
         &self,
         force: bool,
-        table_info: &StageTableInfo,
         catalog_name: &str,
         database_name: &str,
         table_name: &str,
-        files: &[String],
+        stage_files: &[StageFile],
```
Comment (Xuanwo) [inline src/query/service/src/interpreters/interpreter_copy_v2.rs:108]: Nice!

**Review by Xuanwo — APPROVED** [review]
Great!

**Comment by yufan022** [comment]
A marked improvement！

before:
```
COPY INTO import   FROM 's3://xx/10000-100/'   credentials=(aws_key_id='xx' aws_secret_key='xx')   pattern ='.*[.]tsv'   file_format = (type = 'tsv')   force=true;
Query OK, 0 rows affected (1 min 30.55 sec)
```

after:
```
COPY INTO import   FROM 's3://xx/10000-100/'   credentials=(aws_key_id='xx' aws_secret_key='xx')   pattern ='.*[.]tsv'   file_format = (type = 'tsv')   force=true;
Query OK, 0 rows affected (12.19 sec)
```

**Pull request merged**